internal/backend/runtime/omni/controllers/omni/link_status.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2025 Sidero Labs, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+
+package omni
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/controller/generic"
+	"github.com/cosi-project/runtime/pkg/controller/generic/qtransform"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"go.uber.org/zap"
+
+	"github.com/siderolabs/omni/client/api/omni/specs"
+	siderolinkres "github.com/siderolabs/omni/client/pkg/omni/resources/siderolink"
+	"github.com/siderolabs/omni/internal/backend/runtime/omni/controllers/helpers"
+	"github.com/siderolabs/omni/internal/pkg/siderolink"
+)
+
+// LinkStatusControllerName is the name of the LinkStatusController.
+const LinkStatusControllerName = "LinkStatusController"
```

Comment:
Maybe make this a function like
```suggestion
func LinkStatusControllerName(typ resource.Type) string { return fmt.Sprintf("LinkStatusController[%s]", typ) }
```

Just to ensure that if we later refer to this variable from somewhere else to set it as finalizer or something, we don't introduce a bug.